client/components/QnAs/ModalComp.jsx

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import axios from 'axios';
+import Modal from 'react-modal';
+import SearchQs from './SearchQs';
+import QAList from './QAList';
+
+Modal.setAppElement(document.getElementById('app'));
+
+const customStyles = {
+  content: {
+    top: '50%',
+    left: '50%',
+    right: 'auto',
+    bottom: 'auto',
+    marginRight: '-50%',
+    transform: 'translate(-50%, -50%)',
+  },
+};
+
+class ModalComp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      formBody: '',
+      formNickname: '',
+      formEmail: '',
+      newQphotos: [],
+      showModal: false,
+    };
+
+    this.handleNickmameInput = this.handleNickmameInput.bind(this);
+    this.handleQuestionInput = this.handleQuestionInput.bind(this);
+    this.handleEmailInput = this.handleEmailInput.bind(this);
+    this.testForm = this.testForm.bind(this);
+    this.submit = this.submit.bind(this);
+    this.validateEmail = this.validateEmail.bind(this);
+  }
+
+  handleQuestionInput(e) {
+    this.setState({
+      formBody: e.target.value,
+    });
+    e.preventDefault();
+  }
+
+  handleNickmameInput(e) {
+    this.setState({
+      formNickname: e.target.value,
+    });
+    e.preventDefault();
+  }
+
+  handleEmailInput(e) {
+    this.setState({
+      formEmail: e.target.value,
+    });
+    e.preventDefault();
+  }
```

Comment:
If you wanted you could put these all in one method. 

React docs have a really easy example to follow https://reactjs.org/docs/forms.html#handling-multiple-inputs

Reply:
Mmm, good thinking. Putting that on the to-do list.